backend/azure/dtmb/internal/utils/datastructures.go

```diff
@@ -0,0 +1,144 @@
+package utils
+
+import (
+	"container/list"
+	"sync"
+
+	dtmbprotos "github.com/microsoft/durabletask-go/backend/azure/dtmb/internal/backend/v1"
+)
+
+const (
+	DefaultOrchestrationHistoryCacheSize int = 100
+)
+
+type SyncQueue[T dtmbprotos.ExecuteOrchestrationMessage | dtmbprotos.ExecuteActivityMessage | dtmbprotos.Event] struct {
+	lock  *sync.Mutex
+	items []*T
+}
+
+func NewSyncQueue[T dtmbprotos.ExecuteOrchestrationMessage | dtmbprotos.ExecuteActivityMessage | dtmbprotos.Event]() SyncQueue[T] {
+	return SyncQueue[T]{
+		lock:  &sync.Mutex{},
+		items: []*T{},
+	}
+}
+
+func (q *SyncQueue[T]) Enqueue(item *T) {
+	q.lock.Lock()
+	defer q.lock.Unlock()
+	q.items = append(q.items, item)
+}
+
+func (q *SyncQueue[T]) EnqueueBatch(items []*T) {
```

Comment:
Should this use the `...` signature?

Reply:
Yeah - one method instead of two similar ones. Good idea.